WordPress/Classes/ViewRelated/Blaze Campaigns/BlazeCampaignsViewController.swift

```diff
@@ -47,15 +61,70 @@ final class BlazeCampaignsViewController: UIViewController {
         navigationItem.rightBarButtonItem = dismissButton
     }
 
+    private func setupNoResults() {
+        noResultsViewController.delegate = self
+    }
+
+    private func fetchCampaigns() {
+        isLoading = true
+
+        // FIXME: Call BlazeService
+    }
+
     @objc private func promotePostButtonTapped() {
         // TODO: Track event
         BlazeFlowCoordinator.presentBlaze(in: self, source: .campaignsList, blog: blog)
     }
 }
 
+// MARK: - No results
+
+extension BlazeCampaignsViewController: NoResultsViewControllerDelegate {
+
+    private func showNoResultsViewIfNeeded() {
+        guard !isLoading else {
+            showLoadingView()
+            return
+        }
+
+        // FIXME: if results aren't empty, hide the no results view and return
+
+        showNoResultsView()
+    }
+
+    private func showNoResultsView() {
+        hideNoResults()
+        noResultsViewController.hideImageView()
+        configureAndDisplayNoResults(on: view,
+                                     title: NoResults.emptyTitle,
+                                     subtitle: NoResults.emptySubtitle,
+                                     buttonTitle: Strings.promoteButtonTitle)
+    }
+
+    private func showLoadingView() {
+        hideNoResults()
+        configureAndDisplayNoResults(on: view,
+                                     title: NoResults.loadingTitle,
+                                     accessoryView: NoResultsViewController.loadingAccessoryView())
+    }
+
+    func actionButtonPressed() {
+        promotePostButtonTapped()
+    }
+}
+
+// MARK: - Constants
+
 extension BlazeCampaignsViewController {
 
     private enum Strings {
         static let navigationTitle = NSLocalizedString("blaze.campaigns.title", value: "Blaze Campaigns", comment: "Title for the screen that allows users to manage their Blaze campaigns.")
+        static let promoteButtonTitle = NSLocalizedString("blaze.campaigns.promote.button.title", value: "Promote", comment: "Button title for the button that shows the Blaze flow when tapped.")
+    }
+
+    private enum NoResults {
```

Comment:
(Optional) I would suggest nesting it inside `Strings`: `Strings.NoResults.loadingTitle`.